R/connection.R

```diff
@@ -149,3 +151,25 @@ id <- function(db_table_id, conn = NULL, allow_table_only = TRUE) {
 
   return(DBI::Id(schema = db_schema, table = db_table))
 }
+
+#' @importFrom rlang %||%
+#' @export
+id.tbl_dbi <- function(db_table_id, conn = NULL, allow_table_only = TRUE) {
```

Comment:
Did you want to have independent function signatures for the `id` S3 methods?
I.e., did you want to remove the "allow_table_only" argument for every S3 method except `id.character`?